### PR TITLE
Sandbox: add helm config files for adding vtworker

### PR DESCRIPTION
**Pull request by wangyipei01**
but kubectl get pods --namespace still no vtworker process running.

### CODE REVIEW

File: helm/vitess/values.yaml [hunk 1]
```diff
@@ -31,6 +31,8 @@ topology:
         - name: "test_keyspace"
           shards:
             - name: "0"
+              vtworker:
```
Comment (thompsonja) [inline helm/vitess/values.yaml]: This should be at the same level as vtctld.

**Comment by thompsonja** [comment]
In general the piece you are missing is the update to vitess_kubernetes_sandbox.py which generates the values.yaml file. You can see here (https://github.com/youtube/vitess/blob/master/test/cluster/sandbox/vitess_kubernetes_sandbox.py#L128) that if you add another tuple for vtworker it should work. You can also add the logic to override the image if you like.

**Comment by thompsonja** [comment]
I believe you will also need to update vitess.yaml to include the vtworker templates: https://github.com/youtube/vitess/blob/master/helm/vitess/templates/vitess.yaml

File: helm/vitess/templates/_vtworker.tpl [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          securityContext:
+            runAsUser: 999
+          command:
+            - bash
+            - "-c"
+            - |
+              set -ex
+              eval exec /vt/bin/vtworker $(cat <<END_OF_COMMAND
+                -cell={{$cell.name | quote}}
+		-log_dir="$VTDATAROOT/tmp"
```
Comment (thompsonja) [inline helm/vitess/templates/_vtworker.tpl]: There are some tab characters in this file which will cause some problems for you.

**Comment by thompsonja** [comment]
You also need to add some logic in the sandbox code to add vtworker replicas to the generated helm config. https://github.com/youtube/vitess/blob/master/test/cluster/sandbox/vitess_kubernetes_sandbox.py#L184

Add a line like:
vtworker=dict(replicas=self.app_options.vtworker_count),
And also add vtworker_count to the example_sandbox.yaml configurations.

Note that values.yaml is just an example config, used when manually running helm. The sandbox writes its own values.yaml equivalent, so the values.yaml file in helm/vitess will not be used when you run a sandbox.

File: helm/vitess/templates/_vtworker.tpl [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        component: vtworker
+        app: vitess
+      annotations:
+        pod.beta.kubernetes.io/init-containers: '[
+{{ include "init-vtdataroot" (.image | default $0.image) | indent 10 }},
+          {
+            "name": "init-vtworker",
+            "image": {{.image | default $0.image | quote}},
+            "imagePullPolicy": "IfNotPresent",
+            "volumeMounts": [
```
Comment (thompsonja) [inline helm/vitess/templates/_vtworker.tpl]: I don't think you need this volume mount.

File: helm/vitess/templates/_vtworker.tpl [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+              path: /debug/vars
+              port: 17032
+            initialDelaySeconds: 30
+            timeoutSeconds: 5
+          volumeMounts:
+            - name: syslog
+              mountPath: /dev/log
+            - name: vtdataroot
+              mountPath: /vt/vtdataroot
+            - name: web
```
Comment (thompsonja) [inline helm/vitess/templates/_vtworker.tpl]: Similarly, there are no web/certs volume mounts so I think you can delete these too.

**Comment by sougou** [comment]
PR is too old and likely obsolete due to new work. Will be closing this.

**Pull request closed without merging**